samples/cellular/modem_shell/src/ppp/ppp_ctrl.c

```diff
@@ -173,26 +205,33 @@ static void ppp_ctrl_net_mgmt_event_handler(struct net_mgmt_event_callback *cb,
 					    uint32_t mgmt_event,
 					    struct net_if *iface)
 {
-	if ((mgmt_event & (NET_EVENT_PPP_CARRIER_ON |
-			   NET_EVENT_PPP_CARRIER_OFF)) != mgmt_event) {
+	if (mgmt_event == NET_EVENT_PPP_PHASE_RUNNING) {
+		ppp_prot_running = true;
+		mosh_print("PPP in running phase");
 		return;
 	}
+	if (mgmt_event == NET_EVENT_PPP_PHASE_DEAD) {
+		mosh_print("PPP in dead phase");
+	}
 
 	if (mgmt_event == NET_EVENT_PPP_CARRIER_ON) {
 		mosh_print("PPP carrier ON");
 		return;
 	}
-
 	if (mgmt_event == NET_EVENT_PPP_CARRIER_OFF) {
 		mosh_print("PPP carrier OFF");
+		return;
+	}
 
+	if (mgmt_event == NET_EVENT_PPP_PHASE_DEAD && ppp_prot_running) {
```

Comment:
This function could maybe be refactored as a switch-case. Or at least this should be moved inside the `if` on line 213, which already checks if the event is `NET_EVENT_PPP_PHASE_DEAD`.

Reply:
Added a switch-case.